graylog2-server/src/main/java/org/graylog2/plugin/Message.java

```diff
@@ -309,7 +317,7 @@ public void addField(final String key, final Object value) {
         } else if(value instanceof String) {
             final String str = ((String) value).trim();
 
-            if(!str.isEmpty()) {
+            if (isRequiredField || !str.isEmpty()) {
```

Comment:
Yes, you are right, it adds another consistency to keep the old behaviour, that is, messages created with the `Message(final String message, final String source, final DateTime timestamp)` constructor will contain at least  the `message`, and `source` fields, even if their values are empty (or spaces).

I added the special case to fix some tests, but maybe it doesn't make sense. How do you think we should handle the case when a message is created with empty `message` or `source` fields?